src/codewhisperer/util/codeWhispererSession.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const performance = globalThis.performance ?? require('perf_hooks').performance
+
+export class CodeWhispererSession {
+    static #instance: CodeWhispererSession
+
+    // Some other variables for client component latency
+    fetchCredentialStartTime = 0
+    sdkApiCallStartTime = 0
+    invokeSuggestionStartTime = 0
+
+    public static get instance() {
+        return (this.#instance ??= new CodeWhispererSession())
+    }
+
+    setFetchCredentialStart() {
+        if (this.fetchCredentialStartTime === 0 && this.invokeSuggestionStartTime !== 0) {
+            this.fetchCredentialStartTime = performance.now()
+        }
+    }
+
+    setSdkApiCallStart() {
+        if (this.sdkApiCallStartTime === 0 && this.fetchCredentialStartTime !== 0) {
```

Comment:
Does this mean `setSdkApiCallStart` can only be invoked once in this singleton? The second time when I invoke `setSdkApiCallStart()`, `this.sdkApiCallStartTime === 0` is no longer true, but I do need a sdk api call start time. 

Rest of the changes looks good.

Reply:
It should be invoked only once in the beginning of a trigger.
The current logic is somehow complicated for me to follow, making me worrying that we may reset the this state when we still need it, espeically when we delay sending some of the metrics at the start of next trigger.

Reply:
>The current logic is somehow complicated for me to follow
Understood. This is the tech debt we had a long time ago.